portal/server/routes/admin/users.js

```diff
@@ -61,22 +61,38 @@ const handleRoles = (roles) => {
 
 // Admin - user create
 router.post('/users/create', async (req, res) => {
-  const { firstname, surname, roles } = req.body;
-
-  if (firstname && surname && roles) {
+  const {
+    firstname,
+    surname,
+    roles,
+    email,
+    bank,
+  } = req.body;
+
+  if (firstname && surname && roles && bank) {
     const { userToken } = requestParams(req);
-    const userToCreate = {
+    const user = {
       ...req.body,
+      username: email,
       roles: handleRoles(req.body.roles),
     };
 
     // inflate the bank object
     const banks = await getApiData(api.banks(userToken), res);
 
-    const selectedBank = banks.find((bank) => bank.id === userToCreate.bank);
-    userToCreate.bank = selectedBank;
-    userToCreate.username = userToCreate.email;
-    const { status, data } = await api.createUser(userToCreate, userToken);
+    // `fi` stands for `financial institution`
+    if (bank === 'all') {
+      user.bank = {
+        id: '*',
+        name: 'All',
+        hasGefAccessOnly: false,
+      };
+    } else {
+      const selectedBank = banks.find((fi) => fi.id === user.bank);
```

Comment:
I think you could use `b` as the variable without an explanation instead of `fi` (which needs a bit of an explanation)

Reply:
SonarCloud is flagging an issue with non descriptive variable declaration.